src/shared/elf-util.c

```diff
@@ -266,6 +266,42 @@ static int thread_callback(Dwfl_Thread *thread, void *userdata) {
         return DWARF_CB_OK;
 }
 
+static void report_module_metadata(StackContext *c, const char *name, JsonVariant *metadata) {
+        assert(c);
+        assert(name);
+        assert(metadata);
+
+        if (!c->f)
+                return;
+
+        const char
+                *build_id = json_variant_string(json_variant_by_key(metadata, "buildId")),
+                *type = json_variant_string(json_variant_by_key(metadata, "type")),
+                *package = json_variant_string(json_variant_by_key(metadata, "name")),
+                *version = json_variant_string(json_variant_by_key(metadata, "version")),
+                *arch = json_variant_string(json_variant_by_key(metadata, "architecture"));
+
+        fprintf(c->f, "Module %s", name);
+        if (package) {
+                /* The version is only meaningful with a package name. Skip the
+                 * version/arch fields if package is unknown. Architecture is
+                 * meaningful even without version, so print that. */
+                fprintf(c->f, " from %s %s", type ?: "package", package);
+
+                char sep1 = '-', sep2 = '.';  /* rpm (and others?) */
+                if (streq_ptr(type, "deb"))
+                        sep1 = sep2 = '_';    /* deb */
+
+                if (version)
+                        fprintf(c->f, "%c%s", sep1, version);
+                if (arch)
+                        fprintf(c->f, "%c%s", sep2, arch);
```

Comment:
i would prefer outputting it in one of the two formats, always. Pick the one that makes more sense in general.

Reply:
Sorry, I don't grok the comment.

Reply:
i wouldn't try to reconstruct the package name 1:1. But just join the name, arch and version in one way, regardless of distro, and stick to it.

Reply:
No, that's much less useful to users than more specialized approaches. Different packaging systems use different separators to distinguish the parts. If you paste the components together using wrong separators, it is not possible to reverse this, because the reverse operation is ambiguous. Thus, we need to do it correctly. The idea is that the user can take this string and use it interact with the package manager, paste it into a bug report, or a package search online. If we mess up the concatenation, all that is lost.

Reply:
On Debian/Ubuntu one wouldn't search with the full string, but only with the package name - see https://packages.ubuntu.com/ and https://packages.debian.org/